app/src/main/java/com/example/babytracking/BreastActivity.kt

```diff
@@ -1,11 +1,134 @@
+    private var cal: Calendar = Calendar.getInstance()
+
 
-class BreastActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.breast_layout)
+
+        PickerSetup.setupDatePicker(this, breast_Date, cal)
+        PickerSetup.setupTimePicker(this, breast_Time, cal)
+        setupBreastSpinner()
+
+        breast_Save_Button.setOnClickListener {
+            saveInput()
+        }
+    }
+
+    private fun setupBreastSpinner() {
+
+        breast_Spinner ?: return
+        val breastSpinnerAdapter = ArrayAdapter.createFromResource(
+            this, R.array.array_breast,
+            android.R.layout.simple_spinner_item
+        )
+
+        breastSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        breast_Spinner.adapter = breastSpinnerAdapter
+
+
+        breast_Spinner.onItemSelectedListener =
+            object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>, view: View, pos: Int, id: Long
+                ) {
+                    if (!TextUtils.isEmpty(parent.getItemAtPosition(pos) as String)) {
+                        breast = breast_Spinner.selectedItem.toString()
+                    }
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+                    breast = getString(R.string.unknown_breast)
+                }
+            }
+
+    }
+
+
+    @Suppress("DEPRECATION")
+    @SuppressLint("NewApi")
+    private fun saveInput() {
+
+        val lengthText: String = breast_Length.text.toString().trim()
+
+        if (lengthText.isEmpty()) {
+            breast_Length.error = "Length of breastfeeding required"
+            breast_Length.requestFocus()
+            return
+        }
+
+        val lengthInt: Int = lengthText.toInt()
+        val sessionQualForBabyText: String = breast_Quality_Baby.text.toString().trim()
+        val sessionQualForMotherText: String = breast_Quality_Mother.text.toString().trim()
+
+        val timestamp: Timestamp = TimeFormatter.makeTimeStamp(cal)
+        val date = Date(timestamp.time)
+
+        if (lengthInt <= 0) {
+            sleep_Length.error = "Time cannot be negative or null"
+            sleep_Length.requestFocus()
+            return
+        } else if (lengthInt > 24) {
+            sleep_Length.error = "Time cannot exceed 24 hours"
+            sleep_Length.requestFocus()
+            return
+        }
+
+        if (sessionQualForBabyText.isEmpty()) {
+            sleep_Quality.error = "Quality of the session for the baby required"
+            sleep_Quality.requestFocus()
+            return
+        }
+
+        if (sessionQualForMotherText.isEmpty()) {
+            sleep_Quality.error = "Quality of the session for the mother required"
+            sleep_Quality.requestFocus()
+            return
+        }
```

Comment:
All of this would be best in a `validate` function which returns false if the data isn't valid.